src/classes/CW_CampaignRewardCompControllerTest.cls

```diff
@@ -0,0 +1,260 @@
+		
+		CW_Reward_Type__c rewardType = TestDataUtil.createRewardType('FreeBet Amount');
+		insert rewardType;
+		
+		CW_Reward_Type__c cashRewardType = TestDataUtil.createRewardType('Cash');
+		insert cashRewardType;
+		
+		CW_Reward_Catalogue__c rewardCat = TestDataUtil.createRewardCatalogue(offerType.Id,rewardType.Id);
+		rewardCat.Library__c = true;
+		insert rewardCat;
+		
+		rewardCat = TestDataUtil.createRewardCatalogue(offerType.Id,cashRewardType.Id);
+		insert rewardCat;
+		
+		CW_Reward_Library__c rewardLib= TestDataUtil.createRewardLibrary(rewardType.Id);
+		insert rewardLib;
+		
+		CW_Campaign_Offer__c campOffer = TestDataUtil.createCampaignOffer(camp.Id, offerType.Id);
+		insert campOffer; 
+		
+		CW_CampaignRewardCompController compCtrl= new CW_CampaignRewardCompController();
+    	compCtrl.campOfferId = campOffer.Id;
+    	compCtrl.campOffer = campOffer;
+    	compCtrl.offerTypeId = offerType.Id;
+    	
+    	compCtrl.addReward();
+    	system.assertEquals(true,compCtrl.NewRecMode);
+    	compCtrl.getRewardTypeOptions();
+
+    	compCtrl.RewardTypeId = rewardType.Id;
+    	compCtrl.getFields();
+    	compCtrl.campOfferOrRewardHasLibrary();
+    	system.assertEquals('OperationalToolFreeBetAmountFieldSet',compCtrl.fieldsetName,'Wrong fieldset');
+    	system.assertEquals(true,compCtrl.hasLibrary,'Library lookup field is not displayed correctly');
+    	
+    	compCtrl.newRec.Reward_Type__c = cashRewardType.Id;
+    	compCtrl.hasLibrary = false;
+    	compCtrl.newRec.Currency__c = 'GBP';
+    	compCtrl.newRec.Amount__c = 200;
+    	compCtrl.SaveNewRecord();
+    	List<CW_Campaign_Reward__c> campRewards = [SELECT Id FROM CW_Campaign_Reward__c WHERE Campaign_Offer__c =: compCtrl.campOfferId];
+    	system.assertEquals(1,campRewards.size(),'Reward not created successfully');
+    	
+    	//create reward without amount and currency required fields
+    	compCtrl.newRec.Reward_Type__c = cashRewardType.Id;
+    	compCtrl.hasLibrary = false;
+    	compCtrl.newRec.Currency__c =null;
+    	compCtrl.newRec.Amount__c = null;
+    	compCtrl.SaveNewRecord();
+    	campRewards = [SELECT Id FROM CW_Campaign_Reward__c WHERE Campaign_Offer__c =: compCtrl.campOfferId AND Reward_Type__c=:cashRewardType.Id];
+    	system.assertEquals(0,campRewards.size());
+    	
+    	//create reward without reward library required fields
+    	compCtrl.rewardTypeId = rewardType.Id;
+    	compCtrl.hasLibrary = compCtrl.campOfferOrRewardHasLibrary();
+    	compCtrl.newRec.Reward_Library__c = null;
+    	compCtrl.SaveNewRecord();
+    	
+    	List<Apexpages.Message> msgs = ApexPages.getMessages();
+    	Boolean isErrorMsgDisplayed = false;
+    	for(Apexpages.Message msg:msgs)
+    	{
+        	if (msg.getDetail().contains(ConstantUtil.REWARD_WITHOUT_LIBRARY))
+        		isErrorMsgDisplayed = true;
+    	}
+    	system.assert(isErrorMsgDisplayed);//Assert the Page Message was Properly Displayed
+    
+	}
+	
+	static testMethod void createRewardLibrary() 
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility